### PR TITLE
Add arm64 travis builds

### PR DESCRIPTION
**Pull request by glevand**
See: https://github.com/coreos/etcd/issues/5054

### CODE REVIEW

File: build [hunk 1]
```diff
@@ -3,14 +3,29 @@
 ORG_PATH="github.com/coreos"
 REPO_PATH="${ORG_PATH}/etcd"
 
+GIT_SHA=`git rev-parse --short HEAD || echo "GitNotFound"`
+
+LINK_OPERATOR="="
+
+: ${ARCH_LIST:="amd64 arm64"}
```
Comment (heyitsanthony) [inline build]: I don't think it's a good idea to build multiple architectures by default like this. What's wrong with `GOARCH=... ./build` for each arch?

Comment (glevand) [inline build]: @heyitsanthony I think its a good idea to build all supported arches by default.  If a user only wants to build a specific set they can set those with ARCH_LIST.

With your suggestion to build all arches the user needs to invoke the build script multiple times.

Comment (heyitsanthony) [inline build]: Yes. I don't think cross-compiling / building all architectures is the typical use case. Do many projects build for all archs in one shot by default?

Comment (glevand) [inline build]: @heyitsanthony Your point is well taken. I would not say many, but some do.
I think the ability to build for all arches is a valuable one though, to be used before commits/PRs say.  Maybe build only for the host arch as default with an option to build ARCH_LIST?

Comment (heyitsanthony) [inline build]: @glevand I think CI would pick it up with targeted builds anyway. I don't think anyone is actually patient enough to regularly run the full test suite before submitting a PR. I'd prefer to hold off on adding an extra option to ./build until it becomes obviously necessary...

File: .travis.yml [hunk 2]
```diff
@@ -30,4 +36,9 @@ install:
  - pushd cmd/ && go get -t -v ./... && popd
 
 script:
- - ./test
+ - >
+        if [ "${TARGET}" == "amd64" ]; then
+                ARCH_LIST="${TARGET}" ./test;
```
Comment (heyitsanthony) [inline .travis.yml]: `GOARCH="${TARGET}" ./test`

File: .travis.yml [hunk 3]
```diff
@@ -30,4 +36,9 @@ install:
...
+        elif [ "${TARGET}" == "arm64" ]; then
+                ARCH_LIST="${TARGET}" ./build;
```
Comment (heyitsanthony) [inline .travis.yml]: GOARCH="${TARGET}" ./build

File: test [hunk 4]
```diff
@@ -17,7 +17,7 @@ GOSIMPLE_UNUSED_PATHS=$(go list ./... | sed -e 's/github.com\/coreos\/etcd\///g'
 # Invoke ./cover for HTML output
 COVER=${COVER:-"-cover"}
 
-GO_BUILD_FLAGS=-a
+GO_BUILD_FLAGS="-a -v"
```
Comment (heyitsanthony) [inline test]: keep as `-a`? otherwise tests will be flooded with build output

**Comment by heyitsanthony** [comment]
overall this seems OK and the CI hit doesn't look too bad. Are there any solid plans for getting arm64 unit/integration tests working with CI yet?

File: test (hunk 4 above)
Comment (glevand) [inline test]: Do you see that as a problem?  Are they used for anything other than to see what failed?

With `-v` go will print the name of each translation unit, so it is easy to see which one failed.

Comment (heyitsanthony) [inline test]: No strong opinion on this. /cc @xiang90

**Comment by glevand** [comment]
Reworked and pushed out an update that just uses GOARCH.

We are working on a local Jenkins installation to run the tests here at Huawei.  It does not seem practical to do them with github travis, as we would need an arm64 execution environment (QEMU for example).  See: #5054

**Comment by heyitsanthony** [comment]
lgtm after squashing. defer to @xiang90

**Comment by glevand** [comment]
OK, by 'squash', did you want me put these into a single patch?

**Comment by heyitsanthony** [comment]
@glevand yes

File: test (hunk 4 above)
Comment (xiang90) [inline test]: fine with me adding -v

**Comment by xiang90** [comment]
lgtm

**Pull request merged**